src/Sidenav/SubMenu.component.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import styled from 'styled-components';
+import { SideNavItemIcon } from './SideNavItemIcon.component';
+import Icon from 'react-icons-kit';
+
+const SSubMenuItem = styled.div`
+  display: flex;
+  flex-direction: column;
+  transition: height 500ms ease-in-out;
+`;
+const SidebarLink = styled(Link)`
+  float: left;
+  text-decoration: none;
+  list-style: none;
+  height: auto;
+  position: relative;
+  margin: 0;
+  display: inline-flex;
+  align-items: center;
+  padding-left: ${props => (props.level === 1 ? '3rem' : '0rem')};
+  border-bottom: '0px';
+  transition: ${props => props.theme.sidenav.navItem.transition};
+  color: ${props =>
+    props.theme.sidenav[`${props.activeItem ? 'active' : ''}navItem`].color};
+  font-weight: ${props =>
+    props.theme.sidenav[`${props.activeItem ? 'active' : ''}navItem`]
+      .fontWeight};
+  cursor: ${props =>
+    props.disabled
+      ? 'not-allowed'
+      : props.theme.sidenav[`${props.activeSideNavItem ? 'active' : ''}navItem`]
+          .cursor};
+   opacity: ${props =>
+    props.disabled
+      ? '.6'
+      : props.theme.sidenav[`${props.activeItem ? 'active' : ''}navItem`]
+          .opacity};
+:hover {
+    background: ${props =>
+      props.level === 1 ? props.theme.sidenav['activenavItem'].background : ''};
+    border-radius: 7px;
+    transition: all 0.3s;
+  }
+  &:before {
+    content: '';
+    display: block;
+    height: 100%;
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: ${props =>
+      props.theme.sidenav[`${props.activeItem ? 'active' : ''}navItem`]
+        .leftBorderWidth};
+  }
+  .custom-icon-svg {
+       path {
+                 fill: ${props =>
+    props.theme.sidenav[`${props.activeItem ? 'active' : ''}navItem`].color};;
+       }
+    }
+`;
+
+const SidebarLabel = styled.span`
+  padding: ${props =>
+    props.level === 1
+      ? props.theme.sidenav.navLabel.padding
+      : props.theme.sidenav.navIcon.padding};
+  margin-left: 0px;
```

Comment:
Is this margin-left needed? span doesn't use margin since it's an inline element, not block